lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1172,7 +1163,10 @@ private function isLimitQuery() : bool
      */
     private function getSQLForInsert() : string
     {
-        return 'INSERT INTO ' . $this->sqlParts['from']['table'] .
+        /** @var string $table */
```

Comment:
Although it's out of scope, but having to make such assertions points to the fact that the builders for each query type should be implemented in separate classes that would expose only the relevant methods and make sure a proper state when `toString()` is called.

Otherwise, all sorts of madness like this are possible:
```php
echo $conn->createQueryBuilder()
    ->from('foo')
    ->values([])
    ->set('x', 'y')

// SELECT  FROM foo
```

Reply:
This crossed my mind as well. A fluent interface would deserve a proper separation for each kind of SQL statement. Let's come back to this in another PR, maybe!